### PR TITLE
#184581318 Setup Unit Testing

### PR DESCRIPTION
**Pull request by edwinnambaje**
#### What does this PR do?
setting up unit testing

#### Description of Task to be completed?
-To ensure the back end has all functionality working as intended for the unit tests

installed the following packages
- mocha
- chai
- chaiHttp
- added semicolons
- removed jest package

#### How should this be manually tested?
after cloning the repo, 
in terminal add `npm install`
To run tests in terminal add `npm test`
and then add `.env` in your repo and assign the variables as there are in `.env.example` file

#### Any background context you want to provide?
n/a

#### What are the relevant pivotal tracker/Trello stories?
https://www.pivotaltracker.com/n/projects/2632830/stories/184581318

### CODE REVIEW

**Comment by INValentin** [comment]
I believe you may probably the missing semicolons part of the feedback from hound bot, other than I don't think using es6 is  a problem. So I believe your pull request should be merged after that!

**Review by blackd44 — APPROVED** [review]
i think except from circleCi, other has no problems

**Review by INValentin — APPROVED** [review]
I personally think you're good to go!

File: package.json [hunk 1]
```diff
@@ -22,10 +22,15 @@
     "@babel/core": "^7.21.0",
     "@babel/node": "^7.20.7",
     "@babel/preset-env": "^7.20.2",
+    "@babel/register": "^7.21.0",
+    "chai": "^4.3.7",
+    "chai-http": "^4.3.0",
     "cors": "^2.8.5",
     "dotenv": "^16.0.3",
     "express": "^4.18.2",
     "jest": "^29.4.3",
```
Comment (Nkbtemmy) [inline package.json]: what does this jest doing here when you have mocha and Chai ??

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (edwinnambaje) [reply package.json]: i have fixed it

**Review by bencyubahiro77 — APPROVED** [review]

**Review by Patienceineza — APPROVED** [review]
LGTM

**Review by nshutijonathan — CHANGES_REQUESTED** [review]
can you please squash your commits into one?

**Review by Patienceineza — APPROVED** [review]
LGTM

**Review by blackd44 — APPROVED** [review]
Looks Good

**Review by Patienceineza — APPROVED** [review]
**LGTM  did great job keep it up**

**Review by blackd44 — APPROVED** [review]
LGTM

**Review by Habihirwe — APPROVED** [review]
LGTM well done

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Pull request merged**